src/fairchem/core/models/hydra.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import torch
+
+from fairchem.core.common.registry import registry
+from fairchem.core.models.base import BaseModel
+
+
+@registry.register_model("base_hydra")
+class BaseHydra(BaseModel):
+    def __init__(
+        self,
+        num_atoms: int,  # not used
+        bond_feat_dim: int,  # not used
+        num_targets: int,  # not used
+        backbone: dict,
+        heads: dict,
+        otf_graph: bool = True,
+    ):
+        super().__init__()
+        self.otf_graph = otf_graph
+
+        backbone_model_name = backbone.pop("model")
+        self.backbone = registry.get_model_class(backbone_model_name)(
+            num_atoms,
+            bond_feat_dim,
+            num_targets,
```

Comment:
+1 on @rayg1234, if it does not cause too many downstream issues, removing all these unused parameters would be great.